presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ScaledWriterRule.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.matching.Captures;
+import com.facebook.presto.matching.Pattern;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.plan.TableWriterNode;
+
+import java.util.Optional;
+
+import static com.facebook.presto.SystemSessionProperties.useHBOForScaledWriters;
+import static com.facebook.presto.sql.planner.plan.Patterns.tableWriterNode;
+import static com.google.common.base.Preconditions.checkState;
+
+public class ScaledWriterRule
```

Comment:
I just merged https://github.com/prestodb/presto/pull/20990 that tracks which optimizers were cost-based and the source of stats used (CBO/HBO).
Can you override functions isCostBased and getStatsSource so this optimizer also gets tracked?

Reply:
Added in a separate PR https://github.com/prestodb/presto/pull/21120